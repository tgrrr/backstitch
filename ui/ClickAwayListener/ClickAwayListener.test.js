import React from 'react'
import renderer from 'react-test-renderer'
import { ClickAwayListener } from '.'


it('ClickAwayListener: default', () => {
    const component = renderer.create(
        <ClickAwayListener>
            <>
                ClickAwayListener
            </>
        </ClickAwayListener>
    )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// TODO: https://material-ui.com/components/click-away-listener/
// export default function ClickAway() {
//     const [open, setOpen] = React.useState(false);

//     const handleClick = () => {
//         setOpen((prev) => !prev);
//     };

//     const handleClickAway = () => {
//         setOpen(false);
//     };

// <ClickAwayListener onClickAway={handleClickAway}>
//   <div className={classes.root}>
//     <button type="button" onClick={handleClick}>
//       Open menu dropdown
//     </button>
//     {open ? (
//       <div>
//         Click me, I will stay visible until you click outside.
//       </div>
//     ) : null}
//   </div>
// </ClickAwayListener>