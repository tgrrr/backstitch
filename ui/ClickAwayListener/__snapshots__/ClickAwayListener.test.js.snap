// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickAwayListener: default 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <div
    className="Clickawaylistener "
    data-testid="Clickawaylistener"
  >
    ClickAwayListener
  </div>
</div>
`;
